x-pack/test/api_integration/apis/ml/jobs/get_groups.ts

```diff
@@ -0,0 +1,105 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import type { Group } from '@kbn/ml-plugin/common/types/groups';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { MULTI_METRIC_JOB_CONFIG, SINGLE_METRIC_JOB_CONFIG } from './common_jobs';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const testSetupJobConfigs = [SINGLE_METRIC_JOB_CONFIG, MULTI_METRIC_JOB_CONFIG];
+
+  const testCalendarsConfigs = [
+    {
+      calendar_id: `test_get_cal_1`,
+      job_ids: ['multi-metric'],
+      description: `Test calendar 1`,
+    },
+    {
+      calendar_id: `test_get_cal_2`,
+      job_ids: [MULTI_METRIC_JOB_CONFIG.job_id, 'multi-metric'],
+      description: `Test calendar 2`,
+    },
+    {
+      calendar_id: `test_get_cal_3`,
+      job_ids: ['brand-new-group'],
+      description: `Test calendar 3`,
+    },
+  ];
+
+  async function runGetGroupsRequest(user: USER, expectedResponsecode: number): Promise<Group[]> {
+    const { body, status } = await supertest
+      .get('/api/ml/jobs/groups')
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS);
+    ml.api.assertResponseStatusCode(expectedResponsecode, status, body);
+
+    return body;
+  }
+
+  const expectedGroups = [
+    {
+      id: 'automated',
+      jobIds: [MULTI_METRIC_JOB_CONFIG.job_id, SINGLE_METRIC_JOB_CONFIG.job_id],
+      calendarIds: [],
+    },
+    {
+      id: 'brand-new-group',
+      jobIds: [],
+      calendarIds: ['test_get_cal_3'],
+    },
+    {
+      id: 'farequote',
+      jobIds: [MULTI_METRIC_JOB_CONFIG.job_id, SINGLE_METRIC_JOB_CONFIG.job_id],
+      calendarIds: [],
+    },
+    {
+      id: 'multi-metric',
+      jobIds: [MULTI_METRIC_JOB_CONFIG.job_id],
+      calendarIds: ['test_get_cal_1', 'test_get_cal_2'],
+    },
+    {
+      id: 'single-metric',
+      jobIds: [SINGLE_METRIC_JOB_CONFIG.job_id],
+      calendarIds: [],
+    },
+  ];
+
+  describe('get groups', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      for (const job of testSetupJobConfigs) {
+        await ml.api.createAnomalyDetectionJob(job);
+      }
+
+      for (const cal of testCalendarsConfigs) {
+        await ml.api.createCalendar(cal.calendar_id, cal);
+      }
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+      await ml.testResources.deleteIndexPatternByTitle('ft_farequote');
```

Comment:
This suite didn't create an index pattern / data view, so we don't need to clean it up.